### ISSUE TITLE
model = MyModel()  model1 = model works fine but model1 = MyModel() notworking ( means model and model1are not same)!!

### ISSUE THREAD

**Issue by kasanadu** [tensorflow/tensorflow#30074]
<em>Please make sure that this is a bug. As per our [GitHub Policy](https://github.com/tensorflow/tensorflow/blob/master/ISSUES.md), we only address code/doc bugs, performance issues, feature requests and build/installation issues on GitHub. tag:bug_template</em>

**System information**
- Have I written custom code (as opposed to using a stock example script provided in TensorFlow):
- OS Platform and Distribution (e.g., Linux Ubuntu 16.04):
- Mobile device (e.g. iPhone 8, Pixel 2, Samsung Galaxy) if the issue happens on mobile device:
- TensorFlow installed from (source or binary):
- TensorFlow version (use command below):
- Python version:
- Bazel version (if compiling from source):
- GCC/Compiler version (if compiling from source):
- CUDA/cuDNN version:
- GPU model and memory:

You can collect some of this information using our environment capture
[script](https://github.com/tensorflow/tensorflow/tree/master/tools/tf_env_collect.sh)
You can also obtain the TensorFlow version with: 1. TF 1.0: `python -c "import
tensorflow as tf; print(tf.GIT_VERSION, tf.VERSION)"` 2. TF 2.0: `python -c
"import tensorflow as tf; print(tf.version.GIT_VERSION, tf.version.VERSION)"`

**Describe the current behavior**

**Describe the expected behavior**

**Code to reproduce the issue**
Provide a reproducible test case that is the bare minimum necessary to generate the problem.

**Other info / logs**
Include any logs or source code that would be helpful to diagnose the problem. If including tracebacks, please include the full traceback. Large logs and files should be attached.

**Comment by kasanadu** [comment]
class MyModel(Model):
  def __init__(self):
    super(MyModel, self).__init__()
    #self.conv1 = Conv2D(32, 3, activation='relu')
    self.conv1 = Conv2D(32, kernel_size=3, activation='relu')
    self.flatten = Flatten()
    self.d1 = Dense(128, activation='relu')
    self.d2 = Dense(10, activation='softmax')

  def call(self, x):
    x = self.conv1(x)
    x = self.flatten(x)
    x = self.d1(x)
    return self.d2(x)


model = MyModel()
model1 = model
 above is creating model1 with same number of layers of model.
but in the following, model1 is coming up with zero layer and model is having 3 layers.
model1 = MyModel()
// thanks in advance  jk

**Comment by nityansuman** [comment]
<img width="1167" alt="Screenshot 2019-06-24 at 2 49 52 PM" src="https://user-images.githubusercontent.com/23133817/60007135-5c079000-968f-11e9-8ee2-9941ceb50426.png">

I am not able to replicate your issue. Please provide more details. Works fine at my end.

**Comment by kasanadu** [comment]
Thank you for your response.
  In the following  model = MyMddel()   ..this is working fine
 I am trying to create  2
 model = MyMddel()
 model1 = MyMddel()
where model is used in training with Data set and weights are stored in
"model"

[image: image.png]

 Storing weights done in the following ... it is working fi
[image: image.png]

After the above step, i did try Load it on "model1"

it is not going through for  "  model = MyModel()   model1= MyModel()"
[image: image.png]
[image: image.png]

 it is going though   for  "  model = MyModel()   model1=model"

[image: image.png]

Thanks a lot in advance
Regards
jk
from BLR

**Comment by nityansuman** [comment]
Cannot see images.

**Comment by kasanadu** [comment]
Thank you for your response.

  In the following
     model = MyMddel()   \\..this is working fine

 I am trying to create  2
 model = MyMddel()
 model1 = MyMddel()

where model is used in training with Data set and weights are stored in
"model"
Storing weights done in the following ... it is working fine

   model.save_weights('initial_weights-jk1.h5')

After the above step, i did try Load it on "model1"

it is not going through for  "  model = MyModel()   model1= MyModel()"

  model1.load_weights('initial_weights-jk1.h5')

 it is going though   for  "  model = MyModel()   model1=model"

     model1.load_weights('initial_weights-jk1.h5')

Best regards
jk

**Comment by mihaimaruseac** [comment]
Duplicate of #30073 ?

**Comment by ravikyram** [comment]
Can you please help us with .py file, so it will be easy for us to reproduce the issue.Thanks!

**Comment by kasanadu** [comment]
Hello Ravi
  pls find  fnmistJune23v2.ipynb
Best Regards
jk
[fnmistJune23v2.ipynb.zip](https://github.com/tensorflow/tensorflow/files/3325574/fnmistJune23v2.ipynb.zip)

**Comment by kasanadu** [comment]
> Can you please help us with .py file, so it will be easy for us to reproduce the issue.Thanks!

hi ..gm..i did send  ipynb file in the above zip file.. pls use it ...

**Comment by ravikyram** [comment]
I have tried executing the code & i got the following result.Please, let us know is this is the expected output?.
![mynodel](https://user-images.githubusercontent.com/51902062/60318287-10efc480-9990-11e9-950d-61fe5ca80879.png)

**Comment by kasanadu** [comment]
> I have tried executing the code & i got the following result.Please, let us know is this is the expected output?.
> ![mynodel](https://user-images.githubusercontent.com/51902062/60318287-10efc480-9990-11e9-950d-61fe5ca80879.png)

Hi Ravi
![SaveLoadv1](https://user-images.githubusercontent.com/5951312/60320097-4a2b3300-9996-11e9-8a3c-4245d00b9ce4.png)

    model is used to training time  and store weights  in model (file)
    model1 is used to load weights from file

      issue is during load..

**Comment by ravikyram** [comment]
Looks like weights have been saved in the model and loading it from model 1. from the above screenshot. Since the weights has not saved to the model1 ,Please try saving the weights to model1 .Please,check. and let me know if my understanding is correct.Thanks!

**Comment by kasanadu** [comment]
> Looks like weights have been saved in the model and loading it from model 1. from the above screenshot. Since the weights has not saved to the model1 ,Please try saving the weights to model1 .Please,check. and let me know if my understanding is correct.Thanks!

good.. model1 also same class ..  it is another new variable of MyModel.
Loading time, there should be provision to load new variable ( which is model1) of same class MyModel
//jk

**Comment by ravikyram** [comment]
> > Looks like weights have been saved in the model and loading it from model 1. from the above screenshot. Since the weights has not saved to the model1 ,Please try saving the weights to model1 .Please,check. and let me know if my understanding is correct.Thanks!
> 
> good.. model1 also same class .. it is another new variable of MyModel.
> Loading time, there should be provision to load new variable ( which is model1) of same class MyModel
> //jk

Yes you are right model1 is also same object class as model. But When we save weights explicitly to model, model1 will not have provision to load of the weights of model. Thanks!

**Comment by ravikyram** [comment]
Automatically closing due to lack of recent activity. Please update the issue when new information becomes available, and we will reopen the issue. Thanks!
